src/dmd/lexer.d

```diff
@@ -141,6 +147,43 @@ class Lexer
         }
     }
 
+    version (DMDLIB)
+    {
+        this(const(char)* filename, const(char)* base, size_t begoffset, size_t endoffset,
+            bool doDocComment, bool commentToken, bool whitespaceToken)
+        {
+            this(filename, base, begoffset, endoffset, doDocComment, commentToken);
+            this.whitespaceToken = true;
+            this.popFront();
```

Comment:
This introduces an inconsistency between both ctors - instances created by the existing ctor yield `Token.init` as their first element

Reply:
Removed call to `popFront()`.